### PR TITLE
POC: Panther geo chart support

### PR DESCRIPTION
**Pull request by xMort**
image

image

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10332/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12322/ : FAILURE in 8m 23s

File: libs/sdk-ui-geo/src/core/geoChart/helpers/geoChart/data.ts [hunk 1]
```diff
@@ -157,31 +218,35 @@ function getBucketItemNameAndDataIndex(dv: DataViewFacade): IGeoData {
     }, {});
 
     // init data
-    const result: IGeoData = {};
+    const result: IGeoDataBuckets = {};
```
Comment (xMort) [inline libs/sdk-ui-geo/src/core/geoChart/helpers/geoChart/data.ts:221]: All of these changes are just reformat after `BucketNames.LATITUDE and BucketNames.LONGITUDE` was added into the `forEach` array.

File: libs/sdk-ui-geo/src/core/geoChart/helpers/geoChart/data.ts [hunk 2]
```diff
@@ -55,15 +63,45 @@ export function getLocation(latlng: string | null): IGeoLngLat | null {
...
+            "UI-SDK: geoChartDataSource - getLocation: invalid latitude/longitude coordinate",
+            numericalCoordinate,
+        );
+        return null;
+    }
+
+    return numericalCoordinate;
+}
+
+interface IGeoDataBuckets {
```
Comment (xMort) [inline libs/sdk-ui-geo/src/core/geoChart/helpers/geoChart/data.ts:84]: New type used only in the data parsing function to transport parsed latitude and longitude from `getBucketItemNameAndDataIndex` function. Data will not be necessary in inner parts of the chart where already parsed latitude and longitude is used (see the comment below) therefore `IGeoData` interface was not "polluted" with these props.

File: libs/sdk-ui-geo/src/core/geoChart/helpers/geoChart/data.ts [hunk 3]
```diff
@@ -55,15 +63,45 @@ export function getLocation(latlng: string | null): IGeoLngLat | null {
     };
 }
 
+export function parseCoordinate(coordinate: string | null): number | null {
```
Comment (xMort) [inline libs/sdk-ui-geo/src/core/geoChart/helpers/geoChart/data.ts:66]: Almost the same method as `getLocation` above, just without parsing of the latitude and longitude from `latitude;longitude` string. Possibly could be consolidated.

File: libs/sdk-ui-geo/src/GeoPushpinChart.tsx [hunk 4]
```diff
@@ -106,14 +127,41 @@ function GeoPushpinChartInner(props: IGeoPushpinChartProps): JSX.Element {
...
+            {...props}
+            {...{ longitude, latitude, size, color, segmentBy, filters, sortBy }}
+        />
+    );
+};
+
+/**
+ * @public
+ */
+export const GeoPushpinChart = (props: IGeoPushpinChartProps) => {
```
Comment (xMort) [inline libs/sdk-ui-geo/src/GeoPushpinChart.tsx:162]: Split into two smaller components so `useResolveValuesWithPlaceholders` hook can be called conditionally based on prop type.

File: libs/sdk-backend-tiger/src/convertors/fromBackend/CatalogConverter.ts [hunk 5]
```diff
@@ -45,7 +45,7 @@ const tigerLabelToDisplayFormMd = (label: JsonApiLabelOutWithLinks): IAttributeD
         const labelBuilder = commonMetadataObjectModifications(label)(
             builder,
         ) as AttributeDisplayFormMetadataObjectBuilder;
-        labelBuilder.displayFormType(convertLabelType(label.attributes?.valueType));
+        labelBuilder.displayFormType(convertLabelType(label.attributes?.valueType, label.id));
```
Comment (xMort) [inline libs/sdk-backend-tiger/src/convertors/fromBackend/CatalogConverter.ts:48]: This is propagated just for POC purposes to change label type for labels based on identity of the expected labels from "mocked" LDM, i.e., model that has data in expected format but do not support new label types yet.

File: libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx [hunk 6]
```diff
@@ -414,14 +419,40 @@ export class PluggableGeoPushpinChart extends PluggableBaseChart {
...
+            )?.ref;
+            const longitudeDfRef = locationItem.displayForms?.find(
+                (displayForm) => displayForm.type === "GDC.geo.pin_longitude",
+            )?.ref;
+            const latitude = isUriRef(latitudeDfRef) ? latitudeDfRef?.uri : latitudeDfRef?.identifier;
+            const longitude = isUriRef(longitudeDfRef) ? longitudeDfRef?.uri : longitudeDfRef?.identifier;
+            return {
+                tooltipText,
+                latitude,
+                longitude,
```
Comment (xMort) [inline libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx:441]: This also causes these two props to be saved in persisted insight. Not sure if this is OK or not.

image

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Pull request closed without merging**